### PR TITLE
add tests for writing large numbers

### PR DESCRIPTION
**Pull request by pjfanning**

### CODE REVIEW

**Comment by cowtowncoder** [comment]
Ok is the idea here to ensure that the new limits do NOT apply to writer side? Or... ?
If so, may want to try to use for multiple backends (char- and byte-based; `Writer` vs `ByteArrayOutputStream`)?

**Comment by pjfanning** [comment]
Yes - the idea is to ensure the read limits and their defaults are not applied to writes.

I can look at adding similar tests for other backends.

File: src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java [hunk 1]
```diff
@@ -183,6 +188,43 @@ public void testWriteLargeIntegerStringWriter() throws Exception
         }
     }
 
+    public void testWriteLargeIntegerDataOutput() throws Exception
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java:191]: Maybe add a brief comment on background, testing that large/long numbers can be written fine, without changing default settings?

File: src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java [hunk 2]
```diff
@@ -183,6 +188,43 @@ public void testWriteLargeIntegerStringWriter() throws Exception
...
+            gen.writeNumber(bigInt);
+            gen.writeEndObject();
+            gen.close();
+
+            JsonFactory jsonFactory = JsonFactory.builder()
+                    .streamReadConstraints(StreamReadConstraints.builder().maxNumberLength(Integer.MAX_VALUE).build())
+                    .build();
+            try(
+                    ByteArrayInputStream bis = new ByteArrayInputStream(out.toByteArray());
+                    DataInputStream dis = new DataInputStream(bis);
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java:214]: Hmm?

Comment (cowtowncoder) [reply src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java:214]: Oh, I see. That's for DataInput testing.

File: src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java (hunk 1 above)
Comment (cowtowncoder) [reply src/test/java/com/fasterxml/jackson/core/write/UTF8GeneratorTest.java:191]: Actually n/m, I can add.

**Review by cowtowncoder — APPROVED** [review]

**Pull request merged**